torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -290,14 +291,19 @@ def collect_fw_donated_buffer_idxs(
 
     storage_refs = set()
     for t in itertools.chain(fw_ins, user_fw_outs, bw_outs):
-        if isinstance(t, FakeTensor):
+        # Only access storage if a tensor has storage (not sparse)
+        if t is not None and isinstance(t, FakeTensor) and not is_sparse_any(t):
             storage_refs.add(StorageWeakRef(t.untyped_storage()))
 
     num_saved_tensor = len(saved_tensors)
     donated_buffer_idxs = []
     for i in range(num_saved_tensor):
         t = saved_tensors[i]
-        if StorageWeakRef(t.untyped_storage()) not in storage_refs:
+        if (
+            t is not None
+            and not is_sparse_any(t)
```

Comment:
special-casing on `is_sparse` is kind of annoying, but we do it in many other places so this seems fine